plugins/modules/aci_tenant_action_rule_profile.py

```diff
@@ -236,6 +374,24 @@ def main():
     name_alias = module.params.get("name_alias")
 
     aci = ACIModule(module)
+
+    # This dict contains the name of the child classes as well as the corresping attribute input (and attribute name if the input is a string)
+    # this dict is deviating from normal child classes list structure in order to determine which child classes should be created, modified, deleted or ignored.
+    child_classes = dict(
+        rtctrlSetComm=dict(attribute_input=module.params.get("set_community")),
+        rtctrlSetDamp=dict(attribute_input=module.params.get("set_dampening")),
+        rtctrlSetNh=dict(attribute_input=module.params.get("set_next_hop"), attribute_name="addr"),
+        rtctrlSetNhUnchanged=dict(attribute_input=module.params.get("next_hop_propagation")),
```

Comment:
I think this class is not present in 4.2 and thus your aci.get_existing() fails with the filter in the request. 

Causing your tests for 4.2 to fail.